querying/project.go

```diff
@@ -0,0 +1,102 @@
+package querying
+
+import (
+	"regexp"
+	"strings"
+	"sync"
+
+	"golang.org/x/exp/maps"
+)
+
+type ProjectCollection struct {
+	Projects []*Project
+	mu       sync.Mutex
+}
+
+type Project struct {
+	Name     string
+	Findings []*Finding
+	Links    map[string]string
+	mu       sync.Mutex
+}
+
+// NewProject returns a new, empty project with no links or findings.
+func NewProject(name string) *Project {
+	return &Project{
+		Name:     name,
+		Findings: []*Finding{},
+		Links:    map[string]string{},
+	}
+}
+
+// NewProjectCollection returns a new, empty ProjectCollection object.
+func NewProjectCollection() *ProjectCollection {
+	return &ProjectCollection{
+		Projects: []*Project{},
+	}
+}
+
+// normalizeProjectName converts a project name to a standard normalized baseline.
+// This means stripping out all characters which are not: Letters, numbers,
+// spaces, hyphens, or underscores. This is done via a regex, which includes the
+// unicode character classes (\p) of `{L}` to represent all letters, and `{N}`
+// to represent all numbers.
+// Once all undesirable characters have been stripped, both spaces and hyphens
+// are converted to underscores, and the resulting string is lower-cased.
```

Comment:
Makes sense to me, one minor enhancement might be to provide an example of normalization, but that's your call